### DISCUSSION TITLE
What other features should you add?

### DISCUSSION THREAD

**Discussion by bdougie** [open-sauced-craftwork/browser-extension#5] [Ideas]
Do you want to try updating this screen with a "login to OpenSauced button" that connects to our auth API? 

Try matching the OpenSauced orange for the button as well. Logos can be found here. https://github.com/open-sauced/assets

<img width="386" alt="Screen Shot 2023-03-07 at 10 46 38 AM" src="https://user-images.githubusercontent.com/5713670/223522221-6ff0893b-4189-441f-95eb-51cee4052725.png">

**Comment by ishaanbedi** [comment]
> Do you want to try updating this screen with a "login to OpenSauced button" that connects to our auth API?

That's a great idea! I'll surely implement this as well.

I also have some ideas in mind:

- I'd like to share the idea to further extend the use-case of API call by utilising the data returned to generate a pop-up `Profile Card`.  For this, we can use name, url, twitter username etc. that is returned from the API and style it beautifully within our vite extension.

- Further, if a user is not a member of OpenSauced community, we can make the extension popup say `invite user to opensauced community` by extracting the Email/Twitter url from their GitHub bio, if any and generating email/tweet templates using conditional statements in the code.

- To save on [API Rate Limiting Rules](https://api.opensauced.pizza/#/ratelimiting), we can preserve the data fetched by api in the local storage in a json format with 7 days retention so that the end-user using the extension also saves on resources and computing power and the server is also not met with overload of requests by any chance. After 7 days when the profile is loaded again, the API will be called again and preserve the data with new updates (if any).

I'd like to get feedback from other community members as well over this!

Thanks & Regards
Ishaan Bedi
(@ishaanbedi)

**Reply by bdougie** [reply]
These are great thoughts. I like the idea of inviting a user to opensauced, also want to point that there needs to be some sort of value in regards inviting a user. We are working on a teams feature where folks can contribute together on projects in cohorts. 

With the new social links feature on Github you can also send a note via lLnkedin in addition to email or twitter. 

I would love to see an implementation of this caching strategy.

**Reply by AyushAgnihotri2025** [reply]
Hey @bdougie, but from my point of view caching can cause issue if a user has deleted his/her account (on OpenSauced) or updated his/her interest. In this case, it will keep showing the old results.

So, I think no such caching feature should be there. What do you think?

**Reply by ishaanbedi** [reply]
> Hey @bdougie, but from my point of view caching can cause issue if a user has deleted his/her account (on OpenSauced) or updated his/her interest. In this case, it will keep showing the old results.
> 
> So, I think no such caching feature should be there. What do you think?

You are right! I have also inspected this thought and in my code, I've made the time of checking a user through API calls from 7 days to a few minutes, so that in case of any error from the client's network or browser stall or any unexpected event, a large number of calls are not made saving on both user's and server's resources. Based on the team's input, this feature can safely be swiped out or modified at any point in time, if required.
Thanks for the input @AyushAgnihotri2025 
Regards
Ishaan Bedi
(@ishaanbedi)

**Reply by bdougie** [reply]
> Hey @bdougie, but from my point of view caching can cause issue if a user has deleted his/her account (on OpenSauced) or updated his/her interest. In this case, it will keep showing the old results.
> 
> 
> 
> So, I think no such caching feature should be there. What do you think?

Account deletion is possible but that is an edge case to be considered but not blocking. Caching can also include cache invalidation as well.

**Comment by Code-Victor** [comment]
I want to give this a try.
Nevertheless, I'm not sure how to; despite checking the API, I'm not sure how to log in. Any assistance?

**Reply by takanome-dev** [reply]
Since we're using supabase, I think you can install the supabase SDK and connect it with opensauced API. Here is an example of how it's used on the hot repo https://github.com/open-sauced/hot/blob/beta/src/hooks/useSupabaseAuth.ts
Checkout this post on how to connect supabase with your extension https://akoskm.com/how-to-connect-browser-extensions-to-supabase

**Reply by bdougie** [reply]
This is something that should be documented in https://docs.opensauced.pizza. If anyone is interested https://github.com/open-sauced/docs.opensauced.pizza. It would need an issue first.

## related: 

Something that could also be documented is how to get a token from https://hot.opensauced.pizza (this should be added to the docs too).

<img width="285" alt="Screen Shot 2023-03-11 at 7 53 18 PM" src="https://user-images.githubusercontent.com/5713670/224523354-584b1e95-b2f6-471a-81b4-09a69650adb5.png">

On https://api.opensauced.pizza, you auth using that that token here

![Screen Shot 2023-03-08 at 5 25 01 PM](https://user-images.githubusercontent.com/5713670/223891211-ef27833f-0c63-4584-be02-1036d142741a.png)

In the auth section you can test authenticated requests as well.

**Reply by doaortu** [reply]
Hi @TAKANOME-DEV , how do I change the redirect URL to my app?
I copy API_KEY and supabase URL from opensauced hot repo above ,
and also I use the same useSupabaseAuth hooks as in the repo.
but everytime I signIn using the supabase, it always redirect to beta.insight.opensauced.pizza . eventhough I changed the redirect url to localhost.
Or am I doing it wrong? can you help me with this?

**Reply by takanome-dev** [reply]
Hey @doaortu, I have tried it and the redirect is not working as expected. There are a lot of issues on [supabase redirect thing](https://github.com/supabase/gotrue-js/issues/321).
I think you can use `useEffect` to redirect to `http://your-app/somewhere` if there is a session

**Reply by doaortu** [reply]
hi, thanks for responding @TAKANOME-DEV ,
I tried to check the session after login but the session is always undefined.
Maybe my approach was wrong.
Are you able to successfully login to OpenSauced from the chrome extension? if so, what approach do you use?
Thanks in advance.

**Reply by takanome-dev** [reply]
Hi @doaortu, sorry for the late response. I'm not doing this assignment but I will try playing around with it to see how to get that working. But no promises :pizza:

**Reply by bdougie** [reply]
Shout out to @diivi for sending this docs PR. https://github.com/open-sauced/docs.opensauced.pizza/pull/62 

Walkthrough for auth set up is live.

**Reply by Mayank-Tripathi32** [reply]
@bdougie @TAKANOME-DEV Hey, I figured out how to get redirect to work but it requires modification to supabase redirects, It needs to be updated like this,

![image](https://user-images.githubusercontent.com/70465598/228655482-24ffdcce-e422-41c6-9a36-430cc178ac4a.png)

This Part: 
![image](https://user-images.githubusercontent.com/70465598/228656128-a98445eb-98d0-446f-923a-468bf89c5dd4.png)

Basically we need to add github url as a redirect url then it will match the url and it will work as expected.

```
 await signIn({
      provider: "github",
      options: {
        redirectTo: "https://github.com/",
        skipBrowserRedirect: true,
      },
```
I have tested it with my own supabase instance and it works perfectly. Need this to be updated before i can make pull request for my extension tho. Thanks

**Reply by takanome-dev** [reply]
I don't know what's been set for redirect URLs in Supabase, maybe @0-vortex can tell us more about it

**Reply by Mayank-Tripathi32** [reply]
well default doesn't matter as long as there is option for "https://github.com/" it will match the route and redirectTo will work normally.

**Reply by 0-vortex** [reply]
Root domain can be non-slashed, tried replacing `/` with `*` but still had to double all endpoints:

<img width="307" alt="Screenshot 2023-03-29 at 22 42 48" src="https://user-images.githubusercontent.com/237133/228662563-b3eb43d5-285d-4599-b738-6ef580c2f2f2.png">

For the site URL, `localhost` won't work in all cases as it is not capable of storing server cookies. What that means is you can test it on a session basis, an extension domain should be set up like `https://insights.opensauced.pizza`

**Reply by Mayank-Tripathi32** [reply]
@0-vortex Since we are building for github, will it be possible to add that as endpoint so we could store cookies in it? Just need to redirect user back to github after login, which isn't possible unless we add that as endpoint for redirect.

**Reply by 0-vortex** [reply]
The `redirectTo` expects SupaBase client auth handlers to be implemented to read the session or refresh token. GitHub is not our domain so https://supabase.com/docs/guides/auth/social-login/auth-github would not be set up there, meaning redirecting to it has to be overloaded in the extension sandbox or window acrobatics. Why not just redirect to localhost with extension previous location string? 😅

**Reply by doaortu** [reply]
Based on supabase redirect urls which @0-vortex provides, maybe we can setup netlify app, and setup redirect there, grab the access_token, and send to the extension.

or, in my case, I end up grab the access token from hot.opensauced.pizza progamatically and send it to the extension.

**Reply by Mayank-Tripathi32** [reply]
same rewrote the code to grab and save token in https://beta.insights.opensauced.pizza/
btw need to implement signout from extension.

**Reply by Pranav-yadav** [reply]
>meaning redirecting to it has to be overloaded in the extension sandbox or window acrobatics. Why not just redirect to localhost with extension previous location string? 😅

Definitely. That seems one of the easy ways to do it.

But, I would say redirecting to extension homepage is also one the easy ways to do it. (on top of what u said), 
WDYT @0-vortex ?
PS: It seems to be working if implemented properly, but may have many quirks initially ;)

**Reply by 0-vortex** [reply]
> Based on supabase redirect urls which @0-vortex provides, maybe we can setup netlify app, and setup redirect there, grab the access_token, and send to the extension.
> 
> 
> 
> or, in my case, I end up grab the access token from hot.opensauced.pizza progamatically and send it to the extension.

Don't go the netlify path if you are looking for resilient/longg-term solution, that is temporary for deploy previews, just lot the github return url in a sandbox variable, redirect to localhost with query string and then back to github, or navigate between insights/hot for the login and extract values from there. 🔥🍕

**Reply by 0-vortex** [reply]
> >meaning redirecting to it has to be overloaded in the extension sandbox or window acrobatics. Why not just redirect to localhost with extension previous location string? 😅
> 
> 
> 
> Definitely. That seems one of the easy ways to do it.
> 
> 
> 
> But, I would say redirecting to extension homepage is also one the easy ways to do it. (on top of what u said), 
> 
> WDYT @0-vortex ?
> 
> PS: It seems to be working if implemented properly, but may have many quirks initially ;)

For login part yes, for logging out it's easier to just click insights/hot logout button 😁

**Comment by asrvd** [comment]
@bdougie 
clicking on `Print auth token` does nothing for me, I want to implement authentication in my extension as well

![image](https://user-images.githubusercontent.com/68690233/224354491-0fad9c7f-ea70-4abc-b16b-e3c5db774a7b.png)

**Reply by 0-vortex** [reply]
check the devtools console, it should be in there 🍕

**Reply by asrvd** [reply]
> check the devtools console, it should be in there 🍕

oh! thanks a lot!

**Reply by 0-vortex** [reply]
np 👍

**Comment by bdougie** [comment]
This came up on student's PR, sharing here so everyone has context.

If you are willing and want to attempt in adding authentication to opensauced in the extension. That was outlined [here](https://github.com/open-sauced-craftwork/browser-extension/discussions/5#discussioncomment-5248248).

## Voting feature idea 
When a user logs in via the extension, they should be able to vote on a repo. Just like hot.opensauced.pizza, which is also open sourced. A vote button should be placed under or near the existing star buttong.

This is ambitious, so do ask questions of the team if you are blocked. Also no worries if this is too much work, but this would be helpful in learning more on your skills and how we would work together for 6 months.

> Just note that the VoteService in api.opensauced.pizza, requires a repo also be existing in our database first. The SubmissionService will need to be leveraged if a repo is not existing. You might need to mocked that or create your own supabase instance to test. The discord will be helpful for debugging.

<img width="632" alt="Screen Shot 2023-03-11 at 6 58 59 AM" src="https://user-images.githubusercontent.com/5713670/224491848-88db7917-1641-4ab0-b3b4-a536807b3aa2.png">

**Comment by Inzamamdev** [comment]
Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/plain". Strict MIME type checking is enforced for module scripts per HTML spec.

got this error on console and also not able to load the dist file any suggestions

**Reply by bdougie** [reply]
This is not enough context. I'd also recommend opening a new discussion with more context or searching stackoverflow for your error message.

**Comment by VarunGitGood** [comment]
Hiii,
So i've implemented the auth feature but i would like some suggestions here with my approach so feel free 

Basically after i've verified the token i store it in localStorage to further use for additional api calls like upvoting repositories and etc
My only concern is the security can you guys help me out here

Thanks!
